src/App.tsx

```diff
@@ -17,18 +30,36 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                setQuery={setQuery}
+                query={query}
+                setTodos={setTodos}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && <Loader />}
+              {!loading
+                && (
+                  <TodoList
+                    todos={todos}
+                    selectedTodo={selectedTodo}
+                    setSelectedTodo={setSelectedTodo}
+                  />
+                )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo
+        && (
+          <TodoModal
+            id={selectedTodo.userId}
+            todo={selectedTodo}
```

Comment:
Do we really need both of these props? Can we get `userId` from `todo` inside the component?